internal/overlord/servstate/handlers.go

```diff
@@ -345,28 +346,36 @@ func (s *serviceData) startInternal() error {
 		// started (previous logs have already been copied).
 		outputIterator = s.logs.HeadIterator(0)
 	}
-	logWriter := servicelog.NewFormatWriter(s.logs, s.config.Name)
+	serviceName := s.config.Name
+	logWriter := servicelog.NewFormatWriter(s.logs, serviceName)
 	s.cmd.Stdout = logWriter
 	s.cmd.Stderr = logWriter
 
 	// Start the process!
-	logger.Noticef("Service %q starting: %s", s.config.Name, s.config.Command)
-	err = s.cmd.Start()
+	logger.Noticef("Service %q starting: %s", serviceName, s.config.Command)
+	err = reaper.StartCommand(s.cmd)
 	if err != nil {
 		if outputIterator != nil {
 			_ = outputIterator.Close()
 		}
 		_ = s.logs.Close()
 		return fmt.Errorf("cannot start service: %w", err)
 	}
+	logger.Debugf("Service %q started with PID %d", serviceName, s.cmd.Process.Pid)
 	s.resetTimer = time.AfterFunc(s.config.BackoffLimit.Value, func() { logError(s.backoffResetElapsed()) })
 
 	// Start a goroutine to wait for the process to finish.
 	done := make(chan struct{})
+	cmd := s.cmd
 	go func() {
-		waitErr := s.cmd.Wait()
+		exitCode, waitErr := reaper.WaitCommand(cmd)
+		if waitErr != nil {
+			logger.Debugf("Service %q unexpected error: %v", serviceName, waitErr)
```

Comment:
Agreed -- I updated this one to `Noticef`. Note that the log in the `else` case I've left as `Debugf` because it gets logged as Notice in almost all cases inside `exited()` (the debug log there is just to help with debugging -- showing the exit code -- for the rare branches where it doesn't).